src/app/services/fractalis.service.ts

```diff
@@ -174,6 +179,45 @@ export class FractalisService {
     this.variablesInvalid = false;
   }
 
+  public validateVariableUploadStatus(variable: Concept): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      this.getLoadedVariables().then(dataObj => {
+        let message = '';
+        if (dataObj) {
+          let fractalisDataList: FractalisData[] = FractalisService.dataObjectToFractalisDataList(dataObj);
+          let fractalisVars = fractalisDataList.filter(data => data.label === variable.code);
```

Comment:
@gijskant this one links the variables that were loaded into fractalis with GB variables based on the concept code (which is a variable label in fractalis). There is no guarantee that the label is unique (though we can assume that the concept code is unique, right?). There is an option to base this on fractalis unique task_id, but then mapping between concepts and task_ids has to be stored here and also there would be a small change in fractalis required to return the task_id in the POST /data call. Do you think this might would be required?

Reply:
Indeed, concept code is better (and unique), probably the only good identifier to use. Using the name makes it hard to check if the variable has already been loaded (name colisions). However, it is not very descriptive. Can we add a legend that adds a descriptive label for the variables?

Reply:
You mean for variables in fractalis or the variables panel in GB? If you meant the panel, both views still display the names of variables, the grouped (categorized) view probably should display also the concept code